js/customize-snapshots.js

```diff
@@ -40,10 +52,20 @@
 
 			component.extendPreviewerQuery();
 			component.addButtons();
+			component.addSlideDown();
 
 			$( '#snapshot-save' ).on( 'click', function( event ) {
+				var scheduleDate;
 				event.preventDefault();
-				component.sendUpdateSnapshotRequest( { status: 'draft' } );
+				if ( $( this ).html() === component.data.i18n.scheduleButton && ! _.isEmpty( component.snapshotScheduleBox ) && component.getDateFromInputs() && component.isScheduleDateFuture() ) {
```

Comment:
`snapshotStatus` can be `draft` or `future` we can remove that check as we are checking for the future date on the same condition. And will also add `isSnapshotHasUnsavedChanges` check.